app/controllers/pieces_controller.rb

```diff
@@ -1,14 +1,27 @@
 class PiecesController < ApplicationController
-	def show 
+	def edit 
 		@piece = Piece.find(params[:id])
 	end
 
 	def update
 		@piece = Piece.find(params[:id])
-		@game = Game.find(params[:id])
-		if @piece.valid_move?([:x, :y]) && @game(piece_in_space?(x, y) == false)
-			@piece.update_attributes(position_x: x, position_y: y)
-		end	
-		redirect_to game_path(@game)
+		position_x = params[:position_x]
+		position_y = params[:position_y]
+
+		if !current_user_turn 
+			render text: 'Please wait your turn',
+			status: :unauthorized
+		else 
+			@piece.update_attributes(position_x: position_x, position_y: position_y)
```

Comment:
If the move is successful, is there a change to the `turns` column needed to allow the other player to move their piece?

Reply:
To be honest I'm not sure.  I'll talk with Kyle to see if he has any suggestions.

Reply:
Gotcha. I think the white player moves first in chess, so there _may_ be a way to set a default value for that column, then toggle between the two user_ids for a given game